inbox.js

```diff
@@ -0,0 +1,214 @@
+var cron = require('node-cron'),
+    mongoose = require('mongoose'),
+    url = 'mongodb://localhost/EMAILPANEL',
+    conn = mongoose.connect(url),
+    in_array = require('in_array'),
+    fs = require('fs'),
+    Imap = require('imap'),
+    inspect = require('util').inspect;
+
+var google = require('googleapis'),
+    googleAuth = require('google-auth-library'),
+    OAuth2 = google.auth.OAuth2;
+
+var YOUR_CLIENT_ID = "26456531203-aa7qlroeerjrbaoatql4s5di1pcmir46.apps.googleusercontent.com",
+    YOUR_CLIENT_SECRET = "k0jbwm1g1Ow69WiYE_UKu5q4",
+    YOUR_REDIRECT_URL = "https://developers.google.com/oauthplayground";
+
+var oauth2Client = new OAuth2(YOUR_CLIENT_ID, YOUR_CLIENT_SECRET, YOUR_REDIRECT_URL);
+// Retrieve tokens via token exchange explained above or set them:
+oauth2Client.setCredentials({
+    access_token: "ya29.Gls_BEQkhGIpdPplXqN5rUiACQ4spvr27C02bPtHYp1niYkQPIw-6e3EowzetEr-KWEBEtPkZM5J0KJKcbonkf3fTV_XCjvG5EpYVrl0Ev0I0beLdo0fWoYgKoSq",
+    token_type: "Bearer",
+    expires_in: 3600,
+    refresh_token: "1/vjjHYdLPT7STaQEnv_zXPaWTcpR9dg33oiEbk91m8PE"
+});
+var drive = google.drive({ version: 'v2', auth: oauth2Client });
+
+var imap = new Imap({
+    user: 'vaibhav_pathak@excellencetechnologies.in',
+    password: 'VAibhav@1994',
+    host: 'imap.gmail.com',
+    port: 993,
+    tls: true
+});
+var emailSchema = mongoose.Schema({
+    email_id: { type: String, required: true, unique: true },
+    to: { type: String },
+    from: { type: String },
+    sender_mail: { type: String },
+    date: { type: String },
+    email_date: { type: String },
+    email_timestamp: { type: String },
+    subject: { type: String },
+    uid: { type: Number, required: true, unique: true },
+    unread: { type: String },
+    answered: { type: String },
+    body: { type: String },
+    attachment: { type: Array },
+}, {
+    collection: 'emailStored',
+    strict: true,
+});
+var email = conn.model('EMAIL', emailSchema);
+var headers = {};
+var bodyMsg = "";
+
+
+cron.schedule('10 * * * * *', function() {
+    function openInbox(cb) {
+        imap.openBox('INBOX', true, cb);
+    }
+    imap.once('ready', function() {
+        openInbox(function(err, box) {
+            // this will fetch all the Emails data from Gmail
+            var totalmsgs = box.messages.total;
+            var f = imap.seq.fetch("1:10000000", {
```

Comment:
for finding the the all emails upto 10000000    . this process is limitbase another process use ' * ' for fetch newestone